packages/replay/src/eventBuffer/EventBufferArray.ts

```diff
@@ -0,0 +1,75 @@
+import type { ReplayRecordingData } from '@sentry/types';
+
+import type { EventBuffer, RecordingEvent } from '../types';
+
+interface EventsGroup {
+  checkoutTimestamp: number;
+  events: RecordingEvent[];
+}
+
+/**
+ * A basic event buffer that does not do any compression.
+ * Used as fallback if the compression worker cannot be loaded or is disabled.
+ */
+export class EventBufferArray implements EventBuffer {
+  private _events: EventsGroup[];
+
+  public constructor() {
+    this._events = [];
+  }
+
+  /** @inheritdoc */
+  public get pendingLength(): number {
+    return this.pendingEvents.length;
+  }
+
+  /** @inheritdoc */
+  public get pendingEvents(): RecordingEvent[] {
+    return this._events.reduce((acc, { events }) => [...events, ...acc], [] as RecordingEvent[]);
```

Comment:
It's for sure _slower_, not sure of relevantly so? Happy to hear about other approaches to handle this nicely. I've played with solutions that keep everything in one array and keeps record of the checkout pos somehow, but that all felt rather brittle in the end (and also made stuff like getting the checkout time etc. way harder).

I'll give it some more thought!

Reply:
I pushed a change where we now keep two lists of events - a flat one + a nested one. Both just keep references to the same events, so I _think_ it may be more efficient - we now only do any "work" on `clear`, where we check which ones to keep, but keep `addEvent` and `flush` as efficient as possible. WDYT?

Reply:
Ah sorry, I was thinking of a different pattern that I've seen in the frontend codebase a lot. Disregard, but will take a look at the new structure